src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs

```diff
@@ -401,27 +402,26 @@ public virtual void AddReference(string referencePath)
 
                     if (reference != null)
                     {
-                        var path = GetReferencePath(reference);
+                        dteOriginalPath = GetReferencePath(reference);
 
                         // If path != fullPath, we need to set CopyLocal thru msbuild by setting Private
                         // to true.
                         // This happens if the assembly appears in any of the search paths that VS uses to
                         // locate assembly references.
                         // Most commonly, it happens if this assembly is in the GAC or in the output path.
                         // The path may be null or for some project system it can be "".
-                        resolvedToGac = (!string.IsNullOrWhiteSpace(path)
-                            && !StringComparer.OrdinalIgnoreCase.Equals(path, assemblyFullPath));
+                        resolvedToPackage = !string.IsNullOrWhiteSpace(dteOriginalPath) && IsSamePath(dteOriginalPath, assemblyFullPath);
```

Comment:
This is not _just_ a whitespace check per se. The check here is whether original reference path is defined and it is the same as the assembly full path.
Perhaps, better idea would be to move the whitespace check into `GetReferencePath` and make it return `null` when the value is empty or _invalid_. We follow the same approach in many other places. Thoughts?